.github/workflows/black.yml

```diff
@@ -0,0 +1,29 @@
+name: Lint with Black
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Black with PRs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check files using the black formatter
+        uses: rickstaa/action-black@v1
```

Comment:
I've never used this github-action, so no big comment her.  However, I've always just installed black and run it myself in the CI environment (along with isort, pylint, flake8, etc, etc).

Reply:
Agreed, was in two minds about this but I think it's best just to use this test to check the code formatting and run the black formatter locally when required.

Reply:
Ya, this in what I do normally.  You can use --check option for black in CI to check the code only (not format it).  It is probably what this GitHub action does behind the scenes.

Reply:
So then you can add some logic to format if run locally and check only if running in CI.  Or else explicitly pass the options in each environment.